bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,409 @@
+                "",  # S3 LastModified
+                f"{canned_acl_expected}",
+                "",  # Actual Canned ACL
+                f"{gloss_id}",
+                f"{video_id}",
+                f"{gloss_public}",
+                f"{video_public}",
+                action,
+            ]
+        )
+
+    # Get S3 object's ACL
+    result = subprocess.run(
+        [
+            AWSCLI,
+            "s3api",
+            "get-object-acl",
+            "--output",
+            "json",
+            "--bucket",
+            AWS_S3_BUCKET,
+            "--key",
+            video_key,
+        ],
+        env=os.environ,
+        shell=False,
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    canned_acl = "unknown"
+    acls_grants_json = json.loads(result.stdout)["Grants"]
+    if len(acls_grants_json) > 1:
+        if (
+            acls_grants_json[0]["Permission"] == "FULL_CONTROL"
+            and acls_grants_json[1]["Permission"] == "READ"
+        ):
+            canned_acl = "public-read"
+    elif acls_grants_json[0]["Permission"] == "FULL_CONTROL":
+        canned_acl = "private"
+
+    # Get S3 object's LastModified date/time
+    result = subprocess.run(
+        [
+            AWSCLI,
+            "s3api",
+            "get-object-attributes",
+            "--object-attributes",
+            "ObjectParts",
+            "--output",
+            "json",
+            "--bucket",
+            AWS_S3_BUCKET,
+            "--key",
+            video_key,
+        ],
+        env=os.environ,
+        shell=False,
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    s3_lastmodified = json.loads(result.stdout)["LastModified"]
+
+    return CSV_DELIMITER.join(
+        [
+            f"{video_key}",
+            f"{gloss_idgloss}",
+            f"{gloss_created_at}",
+            f"{s3_lastmodified}",
+            f"{canned_acl_expected}",
+            f"{canned_acl}",
+            f"{gloss_id}",
+            f"{video_id}",
+            f"{gloss_public}",
+            f"{video_public}",
+            action,
+        ]
+    )
+
+
+# From the keys present in NZSL, get all their S3 information
+# If we are in 'Go' mode, perform actions
+def process_keys(this_all_keys_dict):
+    print(f"Getting detailed S3 data for keys ({AWS_S3_BUCKET}) ...", file=sys.stderr)
+
+    print(build_csv_header())
+
+    for video_key, [
+        key_in_nzsl,
+        key_in_s3,
+        gloss_idgloss,
+        gloss_created_at,
+        gloss_id,
+        video_id,
+        gloss_public,
+        video_public,
+    ] in this_all_keys_dict.items():
+
+        print(
+            build_csv_row(
+                key_in_nzsl,
+                key_in_s3,
+                video_key,
+                gloss_idgloss,
+                gloss_created_at,
+                gloss_id,
+                video_id,
+                gloss_public,
+                video_public,
+            )
+        )
```

Comment:
suggestion: passing the whole "row" here will mean this function doesn't need to know the specifics of the data

... and in turn make things generally easier as this function will no longer need updating whenever the underlying data changes + the `build_csv_row` function can reuse the data structure more for doing things like calling other functions

Reply:
Done, think this was a legacy carry-over.